ccl/changefeedccl: TestChangefeedNemeses failed

ccl/changefeedccl.TestChangefeedNemeses [failed](https://tanzanite.cluster.engflow.com/invocations/default/955edff4-7b18-4473-bc5f-c08956e90f46?testReportRun=3&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [88218a29ed4165d0de823d2c26e4cffe46924d2a](https://github.com/cockroachdb/cockroach/commits/88218a29ed4165d0de823d2c26e4cffe46924d2a):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestChangefeedNemeses (34m57s)
	TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true (34m57s)
	TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true/pubsub (34m57s)

```
Stack: 

```
goroutine 221583434 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestChangefeedNemeses
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedNemeses1802930096
    test_log_scope.go:76: use -show-logs to present logs inline
=== RUN   TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true
=== RUN   TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true/pubsub
    helpers_test.go:886: making server as system tenant
    helpers_test.go:962: making pubsub feed factory
    nemeses_test.go:30: random seed: -1841764589486462332
    nemeses_test.go:37: using legacy schema changer
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedNemeses.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46998

--- a/pkg/ccl/changefeedccl/nemeses_test.go
+++ b/pkg/ccl/changefeedccl/nemeses_test.go
@@ -26,6 +26,9 @@ func TestChangefeedNemeses(t *testing.T) {
 
 	testutils.RunValues(t, "nemeses_options=", cdctest.NemesesOptions, func(t *testing.T, nop cdctest.NemesesOption) {
 		testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+			if nop.EnableSQLSmith {
+				skip.UnderDeadlock(t, "takes too long under deadlock")
+			}
 			rng, seed := randutil.NewPseudoRand()
 			t.Logf("random seed: %d", seed)
 

